mri/reconstruct/fourier.py

```diff
@@ -209,3 +215,221 @@ def adj_op(self, x):
         """
         self.plan.f = x
         return np.copy(self.plan.adjoint()) / np.sqrt(self.plan.M)
+
+
+class Singleton:
+    numOfInstances = 0
+
+    def countInstances(cls):
+        cls.numOfInstances += 1
+
+    countInstances = classmethod(countInstances)
+
+    def getNumInstances(cls):
+        return cls.numOfInstances
+    getNumInstances = classmethod(getNumInstances)
+
+    def __init__(self):
+        self.countInstances()
+
+
+class NUFFT(FourierBase, Singleton):
+    """  N-D non uniform Fast Fourrier Transform class
+    Attributes
+    ----------
+    samples: np.ndarray
+        the mask samples in the Fourier domain.
+    shape: tuple of int
+        shape of the image (necessarly a square/cubic matrix).
+    nufftObj: The pynufft object
+        depending on the required computational platform
+    platform: string, 'cpu', 'mcpu' or 'gpu'
+        string indicating which hardware platform will be used to compute the
+        NUFFT
+    Kd: int or tuple
+        int or tuple indicating the size of the frequency grid, for regridding.
+        if int, will be evaluated to (Kd,)*nb_dim of the image
+    Jd: int or tuple
+        Size of the interpolator kernel. If int, will be evaluated
+        to (Jd,)*dims image
+    """
+    numOfInstances = 0
+
+    def __init__(self, samples, shape, platform='cpu', Kd=None, Jd=None,
+                 n_coils=1, verbosity=0):
+        """ Initilize the 'NUFFT' class.
+        Parameters
+        ----------
+        samples: np.ndarray
+            the mask samples in the Fourier domain.
+        shape: tuple of int
+            shape of the image (necessarly a square/cubic matrix).
+        platform: string, 'cpu', 'mcpu' or 'gpu'
+            string indicating which hardware platform will be used to
+            compute the NUFFT
+        Kd: int or tuple
```

Comment:
Are these names universal? If not, I think we would benefit from having longer variables names.